cpp/src/io/comp/uncomp.cpp

```diff
@@ -419,6 +419,7 @@ gdf_error io_uncompress_single_h2d(const void *src, gdf_size_type src_size, int
             else if (bz_err == 0)
             {
                 uncomp_len = dst_len;
+                dst.resize(uncomp_len);
```

Comment:
const bool last_line_terminated = (h_uncomp_data[h_uncomp_size - 1] == raw_csv->opts.terminator); 
this check would always return false because of padding at the end of the buffer. Because of this, we would get an extra line with bzip2 input